src/Features/CSharp/Portable/UseExpressionBody/AbstractUseExpressionBodyDiagnosticAnalyzer.cs

```diff
@@ -59,44 +60,76 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
 
         internal virtual Diagnostic AnalyzeSyntax(OptionSet optionSet, TDeclaration declaration)
         {
+            // Note: we will always offer to convert a block to an expression-body (and vice versa)
+            // if possible.  All the user preference does is determine if we show them anything in
+            // the UI (i.e. suggestion dots, or a squiggle) to let them know that they can make the
+            // change.  
+            //
+            // This way, users can turn off the option so they don't get notified, but they can still
+            // make the transformation on a case by case basis.
+            //
+            // Note: if we decide to hide any adornments, then we also broaden the location where the
+            // fix is available.  That way the user can go to any place in the member and choose to
+            // convert it.  Otherwise, they'd have no idea what the 'right' location was to invoke
```

Comment:
A little concerned that this will make the lightbulb too noisy, but we'll have to see as we dogfood it.